### ISSUE TITLE
Websocket connection issue

### ISSUE THREAD

**Issue by stanlyDoge** [fabric8io/kubernetes-client#1708]
I have upgraded from 4.3.0 to 4.4.1 but my operator is failing wit these exceptions: https://pastebin.com/7qAwaavp

I have laso tried 4.4.0 with the same result.
Any idea what went wrong?

```
oc version
oc v3.11.82
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://127.0.0.1:8443
kubernetes v1.11.0+d4cacc0
```

**Comment by rohanKanojia** [comment]
which openshift version are you using?

**Comment by rohanKanojia** [comment]
Could you please provide some test code which seems to be failing on this new release?

**Comment by stanlyDoge** [comment]
I will take a look whether I can, but I may be too complicated to get the malfunctioning part.

**Comment by stanlyDoge** [comment]
I did take a look again at this and there is little bit more of desciption.

So when my operator starts, fabric8 does open a websocket
```
WatchConnectionManager:158 - Request{method=GET, url=https://172.30.0.1/apis/mycrd/v1beta1/namespaces/myproject/crdss?watch=true , tags={}}
WatchConnectionManager:166 - WebSocket successfully opened
```
but each time I want to pull data about crd (e.g. `client.services().inNamespace(namespace).withName("service").get()`)
the socket connection fails with  
```
WatchConnectionManager:304 - Submitting reconnect task to the executor
WatchConnectionManager:320 - Scheduling reconnect task
WatchConnectionManager:114 - Connecting websocket ... io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager@63ca5d90
WatchConnectionManager:158 -  Request{method=GET, url=https://172.30.0.1/apis/mycrd/v1beta1/namespaces/myproject/crdss?watch=true , tags={}}
WatchConnectionManager:176 - onFailure - response is null
WatchConnectionManager:114 - Connecting websocket ... io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager@33603116
2019-09-05 12:07:06 WARN  WatchConnectionManager:209 - Exec Failure
java.io.InterruptedIOException: executor rejected
```

Might this be an issue with reusing a websocket? Not much changed there since 4.3.0 where everything wokrs perfect.

**Comment by rohanKanojia** [comment]
Hmm, I see. Would try to reproduce your error using this.

**Comment by rohanKanojia** [comment]
@stanlyDoge: Hi, 

@walnut-tom : We've fixed this and released 4.5.2, please try it out and share feedback. Closing this issue now, feel free to re-open if you still face this.

**Comment by stanlyDoge** [comment]
Works perfectly. Thank you for such a fast solution!
